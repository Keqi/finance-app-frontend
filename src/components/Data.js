import Navbar from './navbar/Navbar';

function Data() {
  return (
    <div className="Data">
      <Navbar/>
      <p>Data</p>
    </div>
  );
}

export default Data;
