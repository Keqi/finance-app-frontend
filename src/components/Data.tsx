import DataHeader from "./data/DataHeader"
import DataRecord from "./data/DataRecord"

import { useEffect, useState } from 'react';

function Data() {
  const [financeRecords, setFinanceRecords] = useState([]);

  const fetchFinanceRecords = async () => {
    await fetch(`${process.env.REACT_APP_BACKEND_HOST}/finance_records`)
      .then(response => {
        return response.json()
      })
      .then(data => {
        setFinanceRecords(data)
      })
  }

  const handleDelete = async (event: React.ChangeEvent<HTMLInputElement>) => {
    event.preventDefault();
    const recordId = event.target.parentElement.parentElement.getAttribute("record-id");

    try {
      let response = await fetch(`${process.env.REACT_APP_BACKEND_HOST}/finance_records/${recordId}`, {
        method: "DELETE",
        headers: { 'Content-Type': 'application/json' },
      });

      if (response.status === 200) {
        fetchFinanceRecords();
      } else {
        console.log("API request failed.")
      }
    } catch (err) {
      console.log(err);
    }
  }

  useEffect(() => {
    fetchFinanceRecords()
  }, [])

  return (
    <div className="Data">
      <table className="table table-bordered mt-4">
        <thead>
          <tr className="text-center">
            <th className="fs-4" scope="col"></th>
            <th className="fs-4" scope="col" colSpan={4}>ETF (USD)</th>
            <th className="fs-4" scope="col" colSpan={4}>Bonds (PLN)</th>
            <th className="fs-4" scope="col" colSpan={3}>Distribution</th>
            <th className="fs-4" scope="col" colSpan={5}>Total PLN</th>
            <th className="fs-4" scope="col" colSpan={1}>Actions</th>
          </tr>
        </thead>
        <tbody>
          <DataHeader/>

          {financeRecords.map((item, index)=>{
            return <DataRecord
                     key={item.id}
                     financeRecord={item}
                     handleDelete={handleDelete}
                   />
          })}
        </tbody>
      </table>
    </div>
  );
}

export default Data;
