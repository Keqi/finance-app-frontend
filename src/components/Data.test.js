import React from 'react';
import ReactDOM from 'react-dom';
import Data from './Data';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Data/>).toJSON();
  expect(tree).toMatchSnapshot();
});