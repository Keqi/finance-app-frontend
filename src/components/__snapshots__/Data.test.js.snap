// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Data"
>
  <table
    className="table table-bordered mt-4"
  >
    <thead>
      <tr
        className="text-center"
      >
        <th
          className="fs-4"
          scope="col"
        />
        <th
          className="fs-4"
          colSpan={4}
          scope="col"
        >
          ETF (USD)
        </th>
        <th
          className="fs-4"
          colSpan={4}
          scope="col"
        >
          Bonds (PLN)
        </th>
        <th
          className="fs-4"
          colSpan={3}
          scope="col"
        >
          Distribution
        </th>
        <th
          className="fs-4"
          colSpan={5}
          scope="col"
        >
          Total PLN
        </th>
        <th
          className="fs-4"
          colSpan={1}
          scope="col"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="text-center"
      >
        <td
          className="fw-bolder"
        >
          Date
        </td>
        <td
          className="fw-bolder"
        >
          Capital
        </td>
        <td
          className="fw-bolder"
        >
          Total
        </td>
        <td
          className="fw-bolder"
        >
          Profit (USD)
        </td>
        <td
          className="fw-bolder"
        >
          Profit %
        </td>
        <td
          className="fw-bolder"
        >
          Capital
        </td>
        <td
          className="fw-bolder"
        >
          Total
        </td>
        <td
          className="fw-bolder"
        >
          Profit (USD)
        </td>
        <td
          className="fw-bolder"
        >
          Profit %
        </td>
        <td
          className="fw-bolder"
        >
          ETF %
        </td>
        <td
          className="fw-bolder"
        >
          Bonds %
        </td>
        <td
          className="fw-bolder"
        >
          USD/PLN
        </td>
        <td
          className="fw-bolder"
        >
          Capital
        </td>
        <td
          className="fw-bolder"
        >
          Total
        </td>
        <td
          className="fw-bolder"
        >
          Profit PLN
        </td>
        <td
          className="fw-bolder"
        >
          Profit %
        </td>
        <td
          className="fw-bolder"
        >
          Accumulated Inflation
        </td>
        <td
          className="fw-bolder"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
